components/proxy/src/main/java/com/hotels/styx/proxy/ProxyServerBuilder.java

```diff
@@ -54,28 +56,29 @@ public HttpServer build() {
         ProxyServerConfig proxyConfig = environment.styxConfig().proxyServerConfig();
         String unwiseCharacters = environment.styxConfig().get(ENCODE_UNWISECHARS).orElse("");
 
-        return new NettyServerBuilderSpec("Proxy", environment.serverEnvironment(),
+        return new NettyServerBuilderSpec("proxy", environment.serverEnvironment(),
                 new ProxyConnectorFactory(proxyConfig, environment.metricRegistry(), environment.errorListener(), unwiseCharacters, this::addInfoHeader))
                 .toNettyServerBuilder(proxyConfig)
                 .httpHandler(httpHandler)
                 // register health check
                 .register(HealthCheckTimestamp.NAME, new HealthCheckTimestamp())
                 .register("errors-rate-500", new ErrorsRateHealthCheck(environment.metricRegistry()))
-                .doOnStartUp(onStartupAction)
+                .beforeStart(beforeStartAction)
+                .configStore(environment.configStore())
                 .build();
     }
 
     private HttpResponse.Builder addInfoHeader(HttpResponse.Builder responseBuilder, HttpRequest request) {
         return responseBuilder.header(styxInfoHeaderName, responseInfoFormat.format(request));
     }
 
-    public ProxyServerBuilder httpHandler(HttpHandler2 httpHandler) {
+    public ProxyServerBuilder httpHandler(Supplier<HttpHandler2> httpHandler) {
         this.httpHandler = httpHandler;
         return this;
```

Comment:
Can you remind me why the `httpHnadler` has to be loaded lazily? To prevent a blocking load of plugins before the admin interface starts?

Reply:
Yes that is the reason.